config/environment.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+module.exports = function(environment) {
+  var results = {
+    APP: {
```

Comment:
I did this wrong, I should have set it on the second argument to this function, `baseConfig` and returned that. I believe that addresses deep merge.